docs/middleware.rst

```diff
@@ -0,0 +1,43 @@
+Middleware
+==========
+Middleware provides additional functionality that can be attached to any existing transport, 
+application or dispatcher worker. For example, middleware could log inbound and outbound messages, decode a unicoded message,
+store delivery reports in a database or modify a message.
+
+Attaching middleware to your worker is fairly straight forward. Just extend your YAML configuration file with lines like:Attaching middleware to your worker is fairly straight forward. Just extend your YAML configuration file with lines like:
+
+ .. sourcecode:: YAML
+    middlewares:
+    - class_path: vumi2.middlewares.unidecoder.Unidecoder
+        enable_for_connectors: ["mc_qa_expressway"]
+        outbound_enabled: true
```

Comment:
Can you change `sourcecode` to `code-block` both here and below, please?

Also, the YAML indentation's wrong -- all the fields below `class_path:` need to line up with `class_path:`. In other places in the docs we indent lists, so let's do that here as well:

```suggestion
.. code-block:: YAML

    middlewares:
      - class_path: vumi2.middlewares.unidecoder.Unidecoder
        enable_for_connectors: ["mc_qa_expressway"]
        outbound_enabled: true
```